command_line/rescale_diamond_anvil_cell.py

```diff
@@ -0,0 +1,302 @@
+    u"""
+    anvil
+        .caption = "Properties of the mounted diamond anvils"
+    {
+        density = 3510
+            .type = float
+            .help = "The density of the anvil material in kg per cubic metre.  "
+                    "The default is the typical density of diamond."
+
+        thickness = 1.5925
+            .type = float
+            .help = "The thickness in mm of each anvil in the pressure cell.\n"
+                    "The default is the thickness of the pressure cells in use on "
+                    "beamline I19 at Diamond Light Source."
+
+        normal = 0, 1, 0
+            .type = floats(size=3)
+            .help = "A 3-vector representing the normal to the anvil surfaces in the "
+                    "laboratory frame when the goniometer is at zero datum, i.e. the "
+                    "axes are all at zero degrees.  The vector may be given "
+                    "un-normalised."
+    }
+
+    output {
+        experiments = None
+            .type = path
+            .optional = True
+            .help = "The output experiment list file name.\n"
+                    "If None, don't output an experiment list file."
+        reflections = corrected.refl
+            .type = path
+            .help = "The output reflection table file."
+        log = dials.%s.log
+            .type = path
+    }
+    """
+    % filename_root
+)
+
+# Get the tabulated NIST mass attenuation coefficient data for carbon.
+carbon_attenuation_data = attenuation_coefficient.get_table("C")
+
+
+def correct_intensities_for_dac_attenuation(
+    elist,  # type: ExperimentList
+    rtable,  # type: flex.reflection_table
+    dac_norm,  # type: Vector
+    thickness,  # type: float
+    density=3.51,  # type: float  # g·cm⁻³
+):  # type: (...) -> None
+    u"""
+    Boost integrated intensities to account for attenuation by a diamond anvil cell.
+
+    Take an experiment list and reflection table containing integrated but unscaled
+    diffraction data, boost the integrated reflection intensities to correct for the
+    estimated attenuation due to the passage of the incident and diffracted beams
+    through the diamond anvils.
+
+    Args:
+        elist:  An experiment list from integrated data.
+        rtable:  A table of integrated reflections.
+        dac_norm:  A 3-vector representing the normal to the anvil surfaces in the
+                   laboratory frame when the goniometer is at zero datum, i.e. the axes
+                   are all at 0°.  The vector is assumed to be normalised.
+        thickness:  The thickness of each diamond anvil (assumed equal).
+        density:  The density of the anvil material in g·cm⁻³
+                  (units chosen for consistency with the NIST tables of X-ray mass
+                  attenuation coefficients, https://dx.doi.org/10.18434/T4D01F).
+                  Defaults to a typical value for synthetic diamond.
+    """
+    # Iterate over all experiments in the experiment list.
```

Comment:
As a motif I find:

```
# do thing
thing.do()
```

redundant 😉 - appreciate useful when mapping out how code will be laid out but not sure useful in final version

Reply:
Yes that's probably exactly how it came to be there.  And I tend to err on the side of leaving comments in rather than preferring comment-free code.  I know the house style is rather the opposite... :-P  This one though, we could probably lose.